### PR TITLE
via @maxlang: UI cleanup

### PR DESCRIPTION
**Pull request by tamird**
precursor to #2700.

Probably best to review commit-by-commit.

### CODE REVIEW

**Comment by bdarnell** [comment]
The "shrinkwrap format changed" commit creates an empty `bower.installed` file. 

Does this imply that we should be using a particular version of npm? If so, mention the current expected version in the readme/contributing.md. Can we install a particular version of npm with shrinkwrap and then use that when it exists in preference to the global version? (like what we do with `glock`)

`ui/README.md` has some overlap with `CONTRIBUTING.md`; we should probably reduce that. I'd put the part about installing the global dependencies (node and npm) in `CONTRIBUTING.md` and then point to `ui/README.md` for the workflow to make changes to the UI.

LGTM

**Comment by tamird** [comment]
cc @petermattis & @tschottdorf  we had to add a commit that updates node to 4.x, which may be relevant to your interests

File: ui/README.md [hunk 1]
```diff
@@ -27,10 +29,32 @@ Before committing, be sure to run `make` to generate a non-debug version of
 will result in wasted time waiting for the build.
 
 ## Dependencies
-Our admin UI is written in Typescript, so compiling it depends on having the
-typescript compiler locally. We also require a typescript linter (tslint).
+Our admin UI is compiled using a collection of tools that depends on
+[nodejs](https://nodejs.org/), so you'll want to have that installed.
```
Comment (petermattis) [inline ui/README.md]: Is it worth mentioning that we do not use nodejs for serving?

Comment (tamird) [inline ui/README.md]: I don't think so; this is just saying "hey, you'll need node installed or things won't work"

**Comment by bdarnell** [comment]
Is node 4.x now required? This deserves to be called out in the `CONTRIBUTING.md`, especially since getting a version that new is apparently non-trivial in some environments. (including mine. I really need to switch from macports to brew one of these days)

**Comment by maxlang** [comment]
Node 4.x probably isn't necessary, but the Docker image was _way_ too old.

**Comment by mrtracy** [comment]
Is this in addition to #2700?  Or does it replace it?

**Comment by tamird** [comment]
@mrtracy this will go in before #2700, which will be rebased to _only_ do the gulp stuff.

**Comment by mrtracy** [comment]
LGTM

**Comment by bdarnell** [comment]
FWIW, I have node v0.12.7 (via macports), which dates all the way back to august of this year but appears to be new enough to still be usable. The docker image previously had v0.10.29, which is an intolerable 15 months old.

**Comment by tamird** [comment]
@bdarnell gotta move fast!

**Pull request merged**